test/unit/math/rev/fun/lbeta_test.cpp

```diff
@@ -0,0 +1,191 @@
+#include <test/unit/math/expect_near_rel.hpp>
+#include <stan/math/prim.hpp>
+#include <gtest/gtest.h>
+#include <limits>
+#include <vector>
+#include <string>
+#include <cmath>
+
+namespace lbeta_test_internal {
+struct TestValue {
+  double x;
+  double y;
+  double val;
+  double dx;
+  double dy;
+};
+
+constexpr double NaN = std::numeric_limits<double>::quiet_NaN();
+// Test values generated in Mathematice, reproducible notebook at
+// https://www.wolframcloud.com/obj/martin.modrak/Published/lbeta.nb
+// Mathematica Code reproduced below for convenience:
```

Comment:
I don't have a strong opinion, but perhaps this should be recorded in the issue rather than committed here.

Reply:
I didn't realize putting it into an issue is an option, that's a good point! Still I slightly prefer having it in code - it can evolve with the code and so it might make sense to version it along the code. And the code might be around longer than GitHub...